### PR TITLE
[Merged by Bors] - feat: new lemma `Option.bind_congr`

### PR DESCRIPTION
**Pull request by Komyyy**
```lean
theorem bind_congr {f g : α → Option β} {x : Option α}
    (h : ∀ a ∈ x, f a = g a) : x.bind f = x.bind g

@[congr]
theorem bind_congr' {f g : α → Option β} {x y : Option α} (hx : x = y)
    (hf : ∀ a ∈ y, f a = g a) : x.bind f = y.bind g :=
```

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Data/Option/Basic.lean [hunk 1]
```diff
@@ -105,6 +105,10 @@ theorem bind_eq_some' {x : Option α} {f : α → Option β} {b : β} :
 
 #align option.bind_eq_none' Option.bind_eq_none'
 
+theorem bind_congr {f g : α → Option β} {x : Option α}
```
Comment (eric-wieser) [inline Mathlib/Data/Option/Basic.lean:108]: I think this should be tagged `@[congr]`, if Lean allows that?

**Review by eric-wieser — APPROVED** [review]
Other than the possibly-missing attribute, this looks good, thanks!

bors d+

File: Mathlib/Data/Option/Basic.lean (hunk 1 above)
Comment (Komyyy) [reply Mathlib/Data/Option/Basic.lean:108]: I think not. If we were to create a congr theorem, the following theorem would be more appropriate.:
```lean
@[congr]
theorem bind_congr' {f g : α → Option β} {x y : Option α} (hx : x = y)
    (hf : ∀ a ∈ y, f a = g a) : x.bind f = y.bind g :=
  hx.symm ▸ bind_congr hf
```

Comment (Komyyy) [reply Mathlib/Data/Option/Basic.lean:108]: `congr` theorem example: [`dite_congr`](https://leanprover-community.github.io/mathlib4_docs/Init/SimpLemmas.html#dite_congr)

Comment (Komyyy) [reply Mathlib/Data/Option/Basic.lean:108]: I've added this new theorem.

**Pull request closed without merging**